node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/AutoscalingMaintainer.java

```diff
@@ -48,17 +53,31 @@ protected void maintain() {
     private void autoscale(ApplicationId application, List<Node> applicationNodes) {
         try (MaintenanceDeployment deployment = new MaintenanceDeployment(application, deployer, nodeRepository())) {
             if ( ! deployment.isValid()) return; // Another config server will consider this application
-            nodesByCluster(applicationNodes).forEach((clusterSpec, clusterNodes) -> {
-                Optional<AllocatableClusterResources> target = autoscaler.autoscale(application, clusterSpec, clusterNodes);
-                target.ifPresent(t -> log.info("Autoscale: Application " + application + " cluster " + clusterSpec +
-                                               " from " + applicationNodes.size() + " * " + applicationNodes.get(0).flavor().resources() +
-                                               " to " + t.nodes() + " * " + t.advertisedResources()));
+            nodesByCluster(applicationNodes).forEach((clusterId, clusterNodes) -> {
+                Optional<AllocatableClusterResources> target = autoscaler.autoscale(clusterNodes);
+
+                Instant lastLogTime = lastLogged.get(new Pair<>(application, clusterId));
+                if (lastLogTime == null || lastLogTime.isBefore(nodeRepository().clock().instant().minus(Duration.ofHours(1)))) {
```

Comment:
Probably want to skip all of this if target is empty?

Reply:
It's slightly more messy, but ok